src/runtime/test/interface-info-test.ts

```diff
@@ -290,4 +290,64 @@ describe('interface', () => {
     assert.isFalse(check('GibsonCandidate'));
     assert.isFalse(check('LesPaulCandidate'));
   });
+
+  it.only('allows type checking with no names', async () => {
+    const manifest = await Manifest.parse(`
+      interface MyInterface
+        in ~aVar *
+        in ~bVar *
+      schema A
+      schema B
+      particle HostedParticle in 'a.js'
+        in A a
+        in B b
+    `);
+    const myInterface = manifest.interfaces.find(i => i.name === 'MyInterface');
+    const myParticle = manifest.particles.find(p => p.name === 'HostedParticle');
+    assert.isTrue(myInterface.restrictType(myParticle));
+
+    console.log(myInterface.handles.map(h => `${h.type.toString()} => ${h.type.toPrettyString()}`));
+    // This outputs: [ '~aVar => B', '~bVar => A' ]
```

Comment:
I have mixed feelings whether we should rely on ordering in the source file. I'm leaning towards not considering this a problem. What do you think Maria?

@shans